air/src/air/trace_info.rs

```diff
@@ -10,23 +10,19 @@ use utils::{ByteReader, ByteWriter, Deserializable, DeserializationError, Serial
 // CONSTANTS
 // ================================================================================================
 
-/// Number of allowed auxiliary trace segments.
-const NUM_AUX_SEGMENTS: usize = 1;
-
 // TRACE INFO
 // ================================================================================================
 /// Information about a specific execution trace.
 ///
-/// Trace info consists of trace layout info, length, and optional custom metadata. Trace layout
-/// specifies the number of columns for all trace segments. Currently, a trace can consist of at
-/// most two segments. Metadata is just a vector of bytes and can store any values up to 64KB in
+/// Trace info consists of the number of columns for all trace segments, trace length, and optional
+/// custom metadata. Currently, a trace can consist of at most two segments: the main segment and
+/// one auxiliary segment. Metadata is just a vector of bytes and can store any values up to 64KB in
 /// size.
 #[derive(Debug, Clone, Eq, PartialEq)]
 pub struct TraceInfo {
     main_segment_width: usize,
-    aux_segment_widths: [usize; NUM_AUX_SEGMENTS],
-    aux_segment_rands: [usize; NUM_AUX_SEGMENTS],
-    num_aux_segments: usize,
+    aux_segment_width: usize,
+    aux_segment_rand: usize,
```

Comment:
nit: we should probably name this `num_aux_segment_rands`.

Reply:
Done. Also renamed TraceInfo::get_aux_segment_rand_elements() -> TraceInfo::get_aux_segment_num_rand_elements()